File: C/src/TekUretim.c
/** 
* @file ./src/TekUretim.cpp
* @description TekUretim yapısının kaynak dosyası
* @course 1. öğretim B grubu
* @assignment 2. Ödev
* @date 17.05.2023
* @author Sude Çakmak sude.cakmak1@ogr.sakarya.edu.tr
*/

#include "TekUretim.h"

TekUretim TekUrtOlustur(int miktar){  //kurucu fonksiyon
	TekUretim this;
	this = (TekUretim)malloc(sizeof(struct TEKURETIM));
	this->super = new_Uretim(miktar);  //kalıtım alınan yapıyı oluştur
	this->super->uret = &uret;
	this->super->kalitimyoket = &TekUretimYoket;
	this->yoket = &TekUretimYoket;
	return this;
}
int uret(const Uretim this, int aktif_populasyon){  //1 ile 10 arasında değer döndüren fonksiyon
	int deger = rand() + (this->baslangic_populasyon + aktif_populasyon)*3;
	deger = abs(deger);
	deger = deger%10;
	if(1<deger<10) return deger;
	else return 3;  //0 ya da 1 değeri gelme ihtimaline karşılık
}
void TekUretimYoket(const TekUretim this){  //yıkıcı fonksiyon
	if(this == NULL) return;
	this->super->yoket(this->super);
	free(this);
}

File: Java/src/odev/TekUretim.java
/**
*
* @author Sude Çakmak / sude.cakmak1@ogr.sakarya.edu.tr
* @since 30.05.2023
* <p>
* Üretim için kullanılan sınıflardan biri (1-10 arası rastgele değer döndürür)
* </p>
*/
package odev;

import java.util.Random;

public class TekUretim extends Uretim {  //Uretim soyut sınıfından kalıtım alıyor
	public TekUretim(int miktar) {
		super(miktar);    //soyut sınıfın yapıcısını çağırıyor
	}
	
	@Override  //soyut sınıftaki metodun gerçeklemesi
	public int Uret() {   //1 ile 10 arasında rastgele değer döndürür
		Random r = new Random();    //döndüreceği değere rastgelelik katmak için
		int rastgele = this.baslangic_populasyon*13;      //bu sınıfa özgü bir tarafı olması için
		int rand = r.nextInt(100);
		rastgele += rand;
		rastgele = (rastgele % 10) ;   // 0 ile 10 arasında sayı gelmesi için (0 dahil 10 dahil değil)
		if(1 < rastgele && rastgele < 10) return rastgele;
		else return 5;  //0 ya da 1 gelme ihtimaline karşılık (1-10 arasında sayı istendiği için 1 ve 10 dahil değil)
	}
}
